### ISSUE TITLE
Nicer time display in `perf` tool

### ISSUE THREAD

**Issue by eliegoudout** [eliegoudout/las-vegas#7]
Current output of `python3 -m perf` -- introduced on `dev` branch in 5ecf178 -- looks like this:
```pycon
Game time for policy 'None' over 100 games:
╭─────────────┬─────────────┬─────────────┬─────────────┬────────────╮
│ Num players │        Mean │         Std │         Min │        Max │
├─────────────┼─────────────┼─────────────┼─────────────┼────────────┤
│           2 │ 0.000787851 │ 4.55363e-05 │ 0.000712454 │ 0.0010466  │
│           3 │ 0.000864026 │ 6.41655e-05 │ 0.000753319 │ 0.00101083 │
│           4 │ 0.00106616  │ 3.5839e-05  │ 0.000989834 │ 0.00114714 │
│           5 │ 0.00102049  │ 8.67554e-05 │ 0.000913194 │ 0.00138938 │
╰─────────────┴─────────────┴─────────────┴─────────────┴────────────╯
```

It would be nicer to have
```pycon
╭─────────────┬────────┬────────┬───────┬────────╮
│ Num players │   Mean │    Std │   Min │    Max │
├─────────────┼────────┼────────┼───────┼────────┤
│           2 │  788us │ 45.5us │ 712us │ 1.05ms │
│           3 │  864us │ 64.2us │ 753us │ 1.01ms │
│           4 │ 1.07ms │ 35.8us │ 990us │ 1.15ms │
│           5 │ 1.02ms │ 86.7us │ 913us │ 1.39ms │
╰─────────────┴────────┴────────┴───────┴────────╯
```

### To close issue

Implement `format_time(t: float) -> str` from `/perf/__init__.py`.

More specifically:
- Number is displayed on at most 4 characters,
- Time units can be `"ns", "us", "ms", "ss", "mn", "hr", "dd", "mm", "yy"`, though nothing higher than seconds -- `"ss"` -- should be used in practise here.

### Following non triviality needs ruling

Should the value `0.9995` be mapped to `"999ms"` or `"1.00ss"`? The latter seems more natural, but precision is lowered to `10ms`.

Open for discussion.

**Comment by csongorpilinszkinagy** [comment]
IMO 1.00ss would be fine. Those who need precision should revert back to the original time format?

**Comment by eliegoudout** [comment]
Ok I agree. Then `main` should also have a way to output the precise results. Option I think about are:
- Add a positionl-only optional argument `out: bool = False`. If `True`, don't display table but return the resulting `NDArray`,
- Add an output mode positional-only argument `mode: str = 'ui'` describing the type of "output" (`'ui'` for display only, `'var'` for outputing the variable with precise timings, `'both'` for both).
- Always output the result (but I'm not a huge fan since it ruins display if user doesn't catch the result with something like `res = main(...)` or `_ = main(...)`.

The second option seems more flexible, but I can't really imagine a setup where the user would need both the visual output **and** the variable output. So it might be an overkill and the first solutions might be lighter/more natural.

**Comment by csongorpilinszkinagy** [comment]
Understood, implementing the first one then…

**Comment by csongorpilinszkinagy** [comment]
Do I need to implement all the units or just up to seconds?

**Comment by eliegoudout** [comment]
I think up to seconds is enough as long as behaviour (which starts changing above `999ss`) is documented or prohibited (with something like `assert duration < 999.5` for example).

Thanks for the effort!

**Comment by eliegoudout** [comment]
I ended up doing it myself. But I did add what you started regarding `out` option.
- `format_time` implemented in dd39c1d,
- `out` option in 888929a with some code cleaning,
- Added you as a contributor in 4a0de54.

Thanks for your interest!

Cheers

**Comment by csongorpilinszkinagy** [comment]
Thank you for giving me feedback. Looking forward to my next contribution.
